faker/providers/internet/__init__.py

```diff
@@ -165,8 +163,12 @@ def domain_word(self,):
     def tld(self):
         return self.random_element(self.tlds)
 
-    def url(self):
-        pattern = self.random_element(self.url_formats)
+    def url(self, schemes=None):
+        if not schemes:
+            schemes = ['http', 'https']
+
+        pattern = '{}{}'.format(self.random_element(schemes), self.random_element(self.url_formats))
```

Comment:
make sure `schemes` is not an empty list before calling `self.random_element(schemes)`